packages/common/src/i18n/format_date.ts

```diff
@@ -101,6 +101,18 @@ export function formatDate(
   return text;
 }
 
+/**
+ * Return a date with full year instead of converting it to 19{yearProvided}, See:
+ * https://github.com/angular/angular/issues/40377
+ */
+function createDate(year: number, month: number, date: number): Date {
+  const newDate = new Date(year, month, date);
+  if (year >= 0 && year <= 99) {
+    newDate.setFullYear(year - (year - (newDate.getFullYear() - 1900)));
```

Comment:
Wouldn't it be just this (year - year = 0)?
```suggestion
    newDate.setFullYear(newDate.getFullYear() - 1900);
```

Reply:
yes it was stupid 😅, done